extensions/all-extensions/pom.xml

```diff
@@ -23,6 +23,25 @@
             <artifactId>mule-module-file</artifactId>
             <version>${project.version}</version>
             <type>zip</type>
+            <exclusions>
```

Comment:
All of that is fine, but what I'm saying is that the problem goes way beyond extensions. It will be very hard for a developer to figure out that the commons-io classes that the IDE and test cases allows to use, will not be visible in runtime.

What I'm saying is that we should:

* Make sure that just because my maven module depend on mule-core, it doesn't get exposed to mule-core dependencies which are not going to be available in runtime
* Improve our unit test infrastructure so that its classloaders are closer to the runtime ones
* Develop a Mojo which automatically adds the exclusions which you're explicitly adding here... Otherwise this doesn't scale